rfcs/experimental/kernel_templates/README.md

```diff
@@ -104,9 +104,17 @@ sycl::event kernel_template (
 ```
 
 `Arg1`, ..., `ArgN` include sequences to process.
-Data must be passed using the same mechanisms described in the
-[documentation on passing data](https://uxlfoundation.github.io/oneDPL/parallel_api/pass_data_algorithms.html#pass-data-to-algorithms)
-for algorithms with device policies.
+These arguments must reference data which is ready for processing;
+any necessary data transfers must be completed in advance or managed automatically.
+Supported data storage types include:
+- [SYCL buffers](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#subsec:buffers).
+- Unified Shared Memory ([USM](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:usm)).
+- `std::vector` with `sycl::usm_allocator`.
+
+Supported data passing mechanisms are listed in
+the [documentation on passing data](https://uxlfoundation.github.io/oneDPL/parallel_api/pass_data_algorithms.html#pass-data-to-algorithms).
+The `oneapi::dpl::begin` and `oneapi::dpl::end` helper functions are not supported;
+however, `sycl::buffer` objects may be passed directly.
```

Comment:
To make it clear I think we need one more disclaimer to the info in the link, assuming I understand the intent appropriately:

```suggestion
Supported data passing mechanisms are listed in
the [documentation on passing data](https://uxlfoundation.github.io/oneDPL/parallel_api/pass_data_algorithms.html#pass-data-to-algorithms).
The `oneapi::dpl::begin` and `oneapi::dpl::end` helper functions are not supported;
however, `sycl::buffer` objects may be passed directly. `std::vector` with host-allocated data are not supported.
```

Reply:
You are right, thanks for finding that. I am going to apply a slightly corrected suggestion:

```suggestion
Supported data passing mechanisms are listed in
the [documentation on passing data](https://uxlfoundation.github.io/oneDPL/parallel_api/pass_data_algorithms.html#pass-data-to-algorithms).
The `oneapi::dpl::begin` and `oneapi::dpl::end` helper functions are not supported;
however, `sycl::buffer` objects may be passed directly.
`std::vector` with host-allocated data is not supported.
```